python/fasttrackml/client.py

```diff
@@ -0,0 +1,272 @@
+
+        :param run_id: Unique identifier for run
+        :param key: Metric name within the run
+
+        :return: A list of :py:class:`mlflow.entities.Metric` entities if logged, else empty list
+
+        .. code-block:: python
+            :caption: Example
+
+            from fasttrackml import FasttrackmlClient
+
+
+            def print_metric_info(history):
+                for m in history:
+                    print(f"name: {m.key}")
+                    print(f"value: {m.value}")
+                    print(f"step: {m.step}")
+                    print(f"timestamp: {m.timestamp}")
+                    print(f"context: {m.context}")
+                    print("--")
+
+
+            # Create a run under the default experiment (whose id is "0"). Since this is low-level
+            # CRUD operation, the method will create a run. To end the run, you'll have
+            # to explicitly end it.
+            client = FasttrackmlClient()
+            experiment_id = "0"
+            run = client.create_run(experiment_id)
+            print(f"run_id: {run.info.run_id}")
+            print("--")
+
+            # Log couple of metrics, update their initial value, and fetch each
+            # logged metrics' history.
+            for k, v in [("m1", 1.5), ("m2", 2.5)]:
+                client.log_metric(run.info.run_id, k, v, step=0, context={"context_key": "context_value"})
+                client.log_metric(run.info.run_id, k, v + 1, step=1, context={"context_key1": "context_value1"})
+                print_metric_info(client.get_metric_history(run.info.run_id, k))
+            client.set_terminated(run.info.run_id)
+
+        .. code-block:: text
+            :caption: Output
+
+            run_id: c360d15714994c388b504fe09ea3c234
+            --
+            name: m1
+            value: 1.5
+            step: 0
+            timestamp: 1603423788607
+            context: {'context_key': 'context_value'}
+            --
+            name: m1
+            value: 2.5
+            step: 1
+            timestamp: 1603423788608
+            context: {'context_key1': 'context_value1'}
+            --
+            name: m2
+            value: 2.5
+            step: 0
+            timestamp: 1603423788609
+            context: {'context_key': 'context_value'}
+            --
+            name: m2
+            value: 3.5
+            step: 1
+            timestamp: 1603423788610
+            context: {'context_key1': 'context_value1'}
+            --
+        """
+        return self._tracking_client.get_metric_history(run_id, key)
+
+    def get_metric_histories(
+        self,
+        experiment_ids: Optional[List[str]] = None,
+        run_ids: Optional[List[str]] = None,
+        metric_keys: Optional[List[str]] = None,
+        index: str = "step",
+        run_view_type: int = ViewType.ACTIVE_ONLY,
+        max_results: int = 10000000,
+        search_all_experiments: bool = False,
+        experiment_names: Optional[List[str]] = None,
```

Comment:
This endpoint now supports a `context` request param for filtering the results, so we should add an optional dict here.

Reply:
added!